### ISSUE TITLE
PhantomRenderer::renderToPng doesn't seem to work

### ISSUE THREAD

**Issue by mattrobenolt** [grafana/grafana#1696]
On our install, we're trying out grafana 2 beta 2 at this point, and direct image url spits back a fully transparent png.

The logs just simply say:

```
2015/04/01 20:06:34 [I] PhantomRenderer::renderToPng url http://grafana2.local.disqus.net:3000/dashboard/solo/ataglance?panelId=1&fullscreen&from=now-1h&to=now&width=1000&height=500
```

Any ideas?

**Comment by torkelo** [comment]
what os are you running grafana on? If ubuntu /debian try:
`sudo apt-get install libfontconfig`  or if it is redhat/centos try this: 
`yum install -y fontconfig freetype freetype-devel fontconfig-devel``

Phantomjs requires some font libs installed, but it is strange, you should get another error I think if that was the case.

**Comment by mattrobenolt** [comment]
```
$ dpkg -l | grep libfontconfig
ii  libfontconfig1                       2.8.0-3ubuntu9.1                    generic font configuration library - runtime
```

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 12.04.5 LTS
Release:	12.04
Codename:	precise
```

Let me know any other information I can give. :)

**Comment by torkelo** [comment]
What does the image look like? completely transparent? no graph title or anything?

**Comment by mattrobenolt** [comment]
Yeah, it's 100% transparent, but the right dimensions.

**Comment by torkelo** [comment]
Try accessing http://grafana2.local.disqus.net:3000/dashboard/solo/ataglance?panelId=1&fullscreen&from=now-1h&to=now&width=1000&height=500  from the grafana server does that port and domain name look ok?

**Comment by mattrobenolt** [comment]
nginx is in front, so port 3000 is bound to localhost. But over port 80, it's fine. It renders the flot version just fine.

**Comment by torkelo** [comment]
Try changing root_url grafana config option to represent your public facing url (in front of nginx )

**Comment by mattrobenolt** [comment]
Ah, that fixed it. So it's attempting to make an internal http request back to itself?

I just changed by `root_url` to `root_url = %(protocol)s://%(domain)s/`.

**Comment by torkelo** [comment]
Yes,  it is spinning up up phantomjs ( a headless browser) to take the picture

**Comment by mattrobenolt** [comment]
Yeah, it makes sense now. :)

With that said, I very frequently get 500s now just by refreshing the same graph over and over. Sometimes it works, sometimes it doesn't.

Logs only say:

```
2015/04/02 17:08:06 [I] Completed /render 500 Internal Server Error in 23.370464ms
```

**Comment by zeridon** [comment]
i've got one more

    2015/06/23 09:54:05 [I] PhantomRenderer::renderToPng url http://localhost:3000/dashboard/solo/db/templated-ap1?panelId=1&fullscreen&from=now-1h&to=now&var-customer=XXXX&var-service=XXXX&var-instance=XXXX&var-multiplier=10&width=1000&height=500
    2015/06/23 09:54:07 [I] Completed /render 401 Unauthorized in 397.95005ms

graphite backend with basic auth

**Comment by torkelo** [comment]
@zeridon using direct access or proxy access?

**Comment by zeridon** [comment]
Tried with both and no difference.

**Comment by torkelo** [comment]
@zeridon can you describe your setup & config? https or http, any proxy infront, what is your root_url? What version of Grafana are you using?

**Comment by zeridon** [comment]
Some more details:
* DS mode: proxy
* URL scheme: https://%(domain)s/ (grafana is behind ssl apache with basic auth)
* version: 2.0.2 (from https://packagecloud.io/grafana/testing/debian/)
* graphite: separate machine behind nginx proxy (with basic auth)

**Comment by torkelo** [comment]
if you use self signed SSL cert you need to use Grafana 2.1 (pre release builds), png rendering does not work in 2.0.2 with self signed certs. 

you say you use this scheme `URL scheme: https://%(domain)s/ (grafana is behind ssl apache with basic auth)`

But your logs say differently  `PhantomRenderer::renderToPng url http://localhost:3000/`   are you sure your root_url setting is set as `https://%(domain)s/`

**Comment by zeridon** [comment]
yes now they are just fixed them and indeed self signed certs seem to be the problem. will try 2.1 (most of my exchange can be ignored for now)

**Comment by Tenzer** [comment]
I experienced the same problem with only getting a transparent PNG back. It was fixed by installing the latest development build which included b2a0ae0. This was due to the Nginx configuration only allowing clients to connect with TLS and not SSLv3.

**Comment by liyichao** [comment]
We use 2.5, and encounter this problem to.
